examples/CXX_host/example.cpp

```diff
@@ -0,0 +1,278 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * For information on the complete list of contributors to the
+ * XCFun library, see: <https://xcfun.readthedocs.io/>
+ */
+
+#include <cmath>
+#include <iostream>
+
+#include <XCFun/xcfun.h>
+
+// This example contains calls to C++ interface routines that are needed to
+// "talk to" the xcfun library and demonstrates how to use them.
+
+// We will compute the kernel for an unpolarized system using total density and
+// the gradient components as the variables. These are linear in the density
+// matrix, which helps the code using the results from xcfun.
+
+
+// we consider only one grid point
+const int num_grid_points = 1;
+
+// we will use XC_N_NX_NY_NZ
+// N: density
+// NX: x-gradient of the density
+// NY: y-gradient of the density
+// NZ: z-gradient of the density
+const int num_density_variables = 4;
+
+// forward declare function
+double derivative(xc_functional &id,
+                  int vector_length,
+                  double density[][num_density_variables][num_grid_points]);
+
+int main(int, char **) {
+  // print some info and copyright about the library
+  // please always include this info in your code
+  std::cout << xcfun_splash() << std::endl;
+
+  // create a new functional
+  // we need this for interacting with the library
+  auto id = xc_new_functional();
+
+  {
+    // in this example we use PBE
+    std::cout << "Setting up PBE" << std::endl;
+    auto ierr = xc_set(id, "pbe", 1.0);
+    if (ierr) std::cout << "functional name not recognized" << std::endl;
+  }
+
+  //----------------------------------------------------------------------------
+  // in the first example we compute the XC energy ("order 0 derivative")
+  {
+    const auto order = 0;
+    // XC_CONTRACTED here has nothing to do with contracted basis sets
+    // it means we will evaluated in the XC_CONTRACTED mode and internally
+    // contract functional derivatives with the density taylor expansion
+    // in other words: we will not have to explicitly assemble/contract partial
+    // derivatives outside of XCFun
+    auto ierr = xc_eval_setup(id, XC_N_NX_NY_NZ, XC_CONTRACTED, order);
+    if (ierr) std::cout << "xc_eval_setup failed" << std::endl;
+
+    auto vector_length = 1 << order; // bit shift to get power of two: 2**order
+    double density[vector_length][num_density_variables][num_grid_points];
+
+    for (auto i = 0; i < num_grid_points; i++) {
+      // we use fantasy values here
+      density[0][0][i] = 1.0; //         n
+      density[0][1][i] = 2.0; // nabla_x n
+      density[0][2][i] = 3.0; // nabla_y n
+      density[0][3][i] = 4.0; // nabla_z n
+    }
+
+    auto res = derivative(id, vector_length, density);
+    std::cout << "The XC energy density is " << res << std::endl;
+
+    // compare with reference
+    auto diff = std::fabs(-0.86494159400066051 - res);
```

Comment:
I agree it's not the most pressing point, but using `std::fabs` rather than `std::abs` really bothers me. Especially because it caused by the dumb macOS clang with `libc++` that craps out on us. [Here](https://github.com/VCVRack/Rack/commit/c092a27f096a1f21319ee84366b50682de99165a) it seems the fix is to additionally include `cstdlib` :man_shrugging: Could you try it out?

Reply:
I feel your pain. I think I tried `cstdlib` at some point, but I will give it another go.

Reply:
All green! :green_salad:

Reply:
Hey :tada: